pageserver/src/tenant/storage_layer/inmemory_layer.rs

```diff
@@ -284,6 +284,15 @@ impl InMemoryLayer {
     ) -> Result<()> {
         trace!("put_value key {} at {}/{}", key, self.timeline_id, lsn);
 
+        if let Value::Image(buf) = val {
+            if buf.len() > MAX_VALUE_SIZE {
+                tracing::warn!(
+                    "Can't put value of size {} above limit {MAX_VALUE_SIZE} for key {key}",
+                    buf.len()
+                );
+            }
+        }
```

Comment:
```suggestion
        #[cfg(feature = "testing")]
        if val.serialized_size() > MAX_VALUE_SIZE {
            tracing::warn!(%key, size=buf.len(), "too large value");
        }
```

serialized_size is from `trait utils::bin_ser::BeSer` -- alternatively we could do the check after serializing it to buffer on line 293.

It's actually interesting we don't reuse the buffer even for different keys in the same batch.

Reply:
Alerting on the serialized size is something to consider still.